main.py

```diff
@@ -33,15 +33,61 @@ def index():
     }
     return jsonify({"documents": payload}), 200
 
-@app.route("/spots")
+
+@app.route("/api/spots", methods=["GET", "POST"])
 def spots():
-    # Get list of all spots in the database
-    spots = list(Spot.objects.all().values())
-    return Response(
-        response=JSONEncoder().encode(spots),
-        status=200,
-        mimetype="application/json"
-    )
+    if request.method == "POST":
+        # Post a spot to the database
+        body = request.json
+        if not "name" in body or not "description" in body:
+            return "Missing parameters", 400
+        spot = Spot(name=body["name"], description=body["description"])
+        spot.save()
+        return Response(
+            response=JSONEncoder().encode(spot.to_son().to_dict()),
+            status=200
+        )
+    else:
+        # Get list of all spots in the database
+        spots = [x for x in Spot.objects.all().values()]
+        return Response(
+            response=JSONEncoder().encode(spots),
+            status=200,
+            mimetype="application/json"
+        )
+
+@app.route("/api/spots/<id>", methods=["GET", "PUT", "DELETE"])
+def spot_by_id(id):
+    print(id)
+    if request.method == "PUT":
+        body = request.json
+        try:
+            spot = Spot.objects.get({"_id": ObjectId(id)})
+        except Spot.DoesNotExist:
+            return jsonify("Spot not found"), 404
```

Comment:
Let's standardize how we return Flask Responses. Either use `return payload, status_code` or `return Response(response=payload, status_code=status_code)`.

Reply:
The reason for this was because jsonify automatically sets the mimetype to json, but can't be modified to use our own encoder. To fix this I'll add a function json_response(payload, status) that returns a response with payload encoded using our encoder and the mimetype set to json, and then modify the routes to always use this for their responses.

Reply:
Ohhh I see. Thanks for the clarification!